python/src/spark_rapids_ml/tree.py

```diff
@@ -547,10 +547,12 @@ def featureImportances(self) -> Vector:
         """Estimate the importance of each feature."""
         return self.cpu().featureImportances
 
-    @property
-    def getNumTrees(self) -> int:
-        """Number of trees in ensemble."""
-        return self.getOrDefault("numTrees")
+    if not TYPE_CHECKING:
```

Comment:
If everything is type ignore, is this still needed?

Reply:
The root cause seems to be latest mypy does not differentiate function definition and property definition (Spark RandomForestModel object has a getNumTree property). 

TYPE_CHECKING is needed to avoid mypy checking this property definition. Otherwise, mypy throws out a "definition incompatible" error. 

Sorry I did not follow "everything is type ignore". May I have more details?

Reply:
But good to add a comment/TBD to indicate this in the code.